go/consensus/tendermint/abci/mux.go

```diff
@@ -412,7 +404,6 @@ func (mux *abciMux) InitChain(req types.RequestInitChain) types.ResponseInitChai
 
 	// Dispatch registered genesis hooks.
 	mux.RLock()
-	defer mux.RUnlock()
```

Comment:
If `hook()` below panics, this will now lead to the mutex being held forever. You can wrap this part in a function and keep using defer, e.g.:
```golang
func() {
    mux.RLock()
    defer mux.RUnlock()

    mux.logger.Debug("Dispatching genesis hooks")
    // ...
}()
```